mindmaps-test/src/test/java/io/mindmaps/core/ConcurrencyTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * MindmapsDB - A Distributed Semantic Database
+ * Copyright (C) 2016  Mindmaps Research Ltd
+ *
+ * MindmapsDB is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MindmapsDB is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MindmapsDB. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package io.mindmaps.core;
+
+import io.mindmaps.Mindmaps;
+import io.mindmaps.MindmapsGraph;
+import io.mindmaps.concept.Entity;
+import io.mindmaps.concept.EntityType;
+import io.mindmaps.concept.RelationType;
+import io.mindmaps.concept.RoleType;
+import io.mindmaps.exception.MindmapsValidationException;
+import org.javatuples.Pair;
+import org.junit.*;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static io.mindmaps.IntegrationUtils.graphWithNewKeyspace;
+import static io.mindmaps.IntegrationUtils.startTestEngine;
+
+/**
+ *
+ */
+public class ConcurrencyTest {
```

Comment:
This is the test with failing issues right ? We need to look at this one together correct ?